traitlets/traitlets.py

```diff
@@ -908,6 +965,32 @@ def unobserve_all(self):
         """Remove all trait change handlers."""
         self._trait_notifiers = {}
 
+    def register_validator(self, handler, name):
```

Comment:
I think @ellisonbg is in favor of allowing functions to be registered as cross-validators.